From 9cfb5300881a1ac65dfedaccd32e73eb8da60b70 Mon Sep 17 00:00:00 2001
From: Matteo Facchinetti <matteo.facchinetti@sirius-es.it>
Date: Wed, 8 Feb 2017 17:18:58 +0100
Subject: [PATCH] backlight: tps65217_bl: fix initialization when backlight
 node is declared in the main dts

If backlight node is placed in the main dts: pdev->dev.of_node == NULL,
so device driver fail because try to see if platform data is provided.
This HACK, provide a way to use backlight node with capemanager
or directly placed in the main dts.

Signed-off-by: Matteo Facchinetti <matteo.facchinetti@sirius-es.it>
---
 drivers/video/backlight/tps65217_bl.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/video/backlight/tps65217_bl.c b/drivers/video/backlight/tps65217_bl.c
index 31a9203..c6706f5 100644
--- a/drivers/video/backlight/tps65217_bl.c
+++ b/drivers/video/backlight/tps65217_bl.c
@@ -337,9 +337,9 @@ static int tps65217_bl_probe(struct platform_device *pdev)
 	struct backlight_properties bl_props;
 	int brightness = 0;
 
-	tps = NULL;
+	tps = dev_get_drvdata(pdev->dev.parent);
 
-	if (pdev->dev.of_node) {
+	if ((pdev->dev.of_node) || (tps->dev->of_node)) {
 		pdata = tps65217_bl_parse_dt(pdev, &tps, &brightness);
 		if (IS_ERR(pdata)) {
 			dev_err(&pdev->dev, "DT parse failed.\n");
@@ -352,9 +352,6 @@ static int tps65217_bl_probe(struct platform_device *pdev)
 		}
 
 		pdata = pdev->dev.platform_data;
-
-		/* get the parent device */
-		tps = dev_get_drvdata(pdev->dev.parent);
 	}
 
 	if (tps == NULL) {
-- 
2.7.4

